src/esploader.ts

```diff
@@ -0,0 +1,1009 @@
+import { ESPError } from "./error";
+import { inflate, deflate } from "pako";
+import ESP32ROM from "./targets/esp32";
+import { Transport } from "./webserial";
+import { ROM } from "./targets/rom";
+import ESP32C3ROM from "./targets/esp32c3";
+import ESP32S3ROM from "./targets/esp32s3";
+import ESP32S2ROM from "./targets/esp32s2";
+import ESP8266ROM from "./targets/esp8266";
+
+const MAGIC_TO_CHIP: { [key: number]: ROM } = {
+  0x00f01d83: new ESP32ROM(),
+  0x6921506f: new ESP32C3ROM(), // ECO 1+2
+  0x1b31506f: new ESP32C3ROM(), //ECO 3
+  0x09: new ESP32S3ROM(),
+  0x000007c6: new ESP32S2ROM(),
+  0xfff0c101: new ESP8266ROM(),
+};
```

Comment:
Why did you change these from dynamic imports to always imported? This means that the main bundle is way large. The web can easily dynamically import things.

If for some reason you do not want this in bundle.js (it used to work there previously), you can use the Rollup option [`inlineDynamicOption`](https://rollupjs.org/guide/en/#outputinlinedynamicimports)

Reply:
I think the size would be the same because the bundle will still include all targets with `inlineDynamicImports`. The alternative would mean to include those targets js files to deployment, which will against the idea of having a bundle single js file. We could probably optimize this later to bundle for specific targets ?

Reply:
The NPM package should include 2 builds. A build that is just the transpiled version. Leave it up to the consumer of this lib to decide how it should deal with bundling of this lib, including de duplication of transpiration artifacts. 

The second build can be a single bundle.js build, which can use inlineDynamicImports to generate a single file. However, I think that’s overkill and will actually hurt performance. Users want to connect fast, and it’s a waste to let their computers download ALL chips (including all future ESP chips!) before they can connect one. 

When do you see the bundle.js being used directly? Because I thought that the goal was to publish it on NPM and then it’s up to the consumer to deal with it?

Reply:
Yes @balloob this is what we have at the moment. NPM package includes the `lib` folder with transpiled JS files with TS types and the `bundle.js` made from those JS files which I think is useful for browser deployment since it is better to have single minified file (yet to minify it xD) like releasing it in unpkg and other CDNs.

Reply:
Unpkg works with modules and relative imports just fine. That’s what we do for ESP Web Tools and it looks like this https://unpkg.com/browse/esp-web-tools@9.0.4/dist/web/ (this split is the result of the dynamic imports in esptool.js)